src/libopensc/card-cac.c

```diff
@@ -1923,6 +1929,7 @@ static struct sc_card_driver * sc_get_driver(void)
 	cac_ops.decipher =  cac_decipher;
 	cac_ops.card_ctl = cac_card_ctl;
 	cac_ops.pin_cmd = cac_pin_cmd;
+	cac_ops.logout = cac_logout;
```

Comment:
For the CAC, there is `card->ops->logout = NULL;` in `cac_match_card()`. I am no longer sure what was the intention of that part initially, but it looks like it makes the changes here no-op?

Very similar code is in muscle, piv, coolkey and cac1 drivers now.

Reply:
I need to check on this...

Reply:
I agree, that the code in match_card resets logout to NULL. A similar check in match_card appeared first in b3822c349b9b4c3e837298cd0d1e333faa09a604 for the muscle driver and seemed to be copied to other drivers later. I don't know what it is trying to achieve, but it looks like that it is not needed anymore.